playbooks/robusta_playbooks/pvc_snapshots.py

```diff
@@ -0,0 +1,55 @@
+from pydantic import SecretStr
+from kubernetes import client
+from robusta.api import *
+
+
+class VolumeSnapshotParams(ActionParams):
+    """
+    :var name: name of the PersistentVolumeClaim to backup
+    :var namespace: namespace of the PersistentVolumeClaim to backup
+    :example name: some_pvc_name
+    """
+
+    name: str
+    namespace: str = "default"
+
+
+@action
+def create_pvc_snapshot(event: ScheduledExecutionEvent, params: VolumeSnapshotParams):
+    """
+    Backup a PersistentVolume by creating a VolumeSnapshot
+    """
+
+    pvc = PersistentVolumeClaim().read(name=params.name, namespace=params.namespace)
+    formatted_date = datetime.now().strftime("%m-%d-%y")
+    snapshot_name = f"{params.name}-backup-{formatted_date}-{event.recurrence}"
+
+    # both the kubernetes client and hikaru don't have proper support for snapshots in the autogenerated classes
+    # hence the workarounds here
+    # see https://stackoverflow.com/questions/67157702/can-i-take-a-volume-snapshot-with-the-k8s-python-client
+    snapshot_body = {
+        "apiVersion": "snapshot.storage.k8s.io/v1beta1",
+        "kind": "VolumeSnapshot",
+        "metadata": {"name": snapshot_name, "namespace": pvc.metadata.namespace},
+        "spec": {
+            # "volumeSnapshotClassName": "example-snapshot-class",
```

Comment:
Did you leave the comment on purpose ?

Reply:
Yes, we're not specifying a snapshot class right now but it's very possible we'll want to in the future.